crates/extensions/tedge_file_system_ext/src/lib.rs

```diff
@@ -47,7 +47,16 @@ impl FsWatchMessageBox {
         log_message_sent("Inotify", &message);
 
         for (watch_path, sender) in self.watch_dirs.iter_mut() {
-            if path.starts_with(watch_path) {
+            // Try to use canonical paths to avoid false negatives when dealing with symlinks
+            if path
+                .canonicalize()
+                .unwrap_or_else(|_| path.to_path_buf())
+                .starts_with(
+                    watch_path
+                        .canonicalize()
+                        .unwrap_or_else(|_| watch_path.to_path_buf()),
+                )
```

Comment:
I would avoid to do this canonicalization for each and every event.

It's better to:
1. canonicalize the `path` outside the loop.
2. canonicalize the `watch_path` when registered.